event_types.go

```diff
@@ -71,26 +71,52 @@ func (event Event) String() string {
 }
 
 // merge combines this event with the other event (assumed to be newer!).
-// TODO: Intelligent merging: if there are multiple EventUpdate(Flags) events, can we just take the latest one?
-func (event *Event) merge(other Event) error {
-	event.EventID = other.EventID
+func (event *Event) merge(next Event) error {
+	event.EventID = next.EventID
 
-	if other.User != nil {
-		event.User = other.User
+	if next.User != nil {
+		event.User = next.User
 	}
 
-	if other.MailSettings != nil {
-		event.MailSettings = other.MailSettings
+	if next.MailSettings != nil {
+		event.MailSettings = next.MailSettings
 	}
 
-	// For now, label events are simply appended.
-	event.Labels = append(event.Labels, other.Labels...)
+	for _, nextLabel := range next.Labels {
+		if nextLabel.Action == EventUpdate {
+			if idx := xslices.IndexFunc(event.Labels, func(old LabelEvent) bool {
+				return old.Action == EventUpdate && old.ID == nextLabel.ID
+			}); idx > 0 {
+				event.Labels = append(xslices.Remove(event.Labels, idx, 1), nextLabel)
```

Comment:
Do you have assurances that removing the event and then leaving the state at the end does not lead to unexpected behavior?

Reply:
I can't think of a reason it would lead to unexpected behaviour 🤔 though i wonder if it would be better to replace the earlier event with the later one, not the other way around.